### ISSUE TITLE
Upgrading through the GUI failed if connection used a file inside the server path

### ISSUE THREAD

**Issue by cmotelet** [SignalK/signalk-server#1170]
When upgrading through the GUI, if a connection used a file inside the server path, npm remove the most part of signalk-server package but failed when try to remove the file opened by himself.
Then npm stop without reinstalling the new version.
When the user tries to restart, the signalk-server package has disappeared and everything is broken.

The problem is found under Windows, I have to check if it also occurs under Linux

```
Oct 25 19:25:23 stderr:
Oct 25 19:25:23 stderr: npm ERR! code
Oct 25 19:25:23 stderr: EPERM
Oct 25 19:25:23 stderr: npm ERR! syscall rename npm ERR! path c:\signalk\nodejs\node_modules\signalk-server npm ERR! dest c:\signalk\nodejs\node_modules\.signalk-server.DELETE
Oct 25 19:25:23 stderr: npm ERR! errno -4048
Oct 25 19:25:23 stderr: npm ERR! Error: EPERM: operation not permitted, rename 'c:\signalk\nodejs\node_modules\signalk-server' -> 'c:\signalk\nodejs\node_modules\.signalk-server.DELETE' npm ERR! { [Error: EPERM: operation not permitted, rename 'c:\signalk\nodejs\node_modules\signalk-server' -> 'c:\signalk\nodejs\node_modules\.signalk-server.DELETE'] npm ERR! cause: npm ERR! { Error: EPERM: operation not permitted, rename 'c:\signalk\nodejs\node_modules\signalk-server' -> 'c:\signalk\nodejs\node_modules\.signalk-server.DELETE' npm ERR! errno: -4048, npm ERR!
Oct 25 19:25:23 stderr: code: 'EPERM', npm ERR! syscall: 'rename', npm ERR! path: 'c:\\signalk\\nodejs\\node_modules\\signalk-server', npm ERR! dest: 'c:\\signalk\\nodejs\\node_modules\\.signalk-server.DELETE' }, npm ERR! stack: npm ERR! 'Error: EPERM: operation not permitted, rename \'c:\\signalk\\nodejs\\node_modules\\signalk-server\' -> \'c:\\signalk\\nodejs\\node_modules\\.signalk-server.DELETE\'', npm ERR! errno: -4048, npm ERR! code: 'EPERM', npm ERR! syscall: 'rename', npm ERR! path: 'c:\\signalk\\nodejs\\node_modules\\signalk-server', npm ERR! dest: 'c:\\signalk\\nodejs\\node_modules\\.signalk-server.DELETE' } npm ERR! npm ERR! The operation was rejected by your operating system. npm ERR!
Oct 25 19:25:23 stderr: It's possible that the file was already in use (by a text editor or antivirus), npm ERR! or that you lack permissions to access it. npm ERR! npm ERR! If you believe this might be a permissions issue, please double-check the npm ERR! permissions of the file and its containing directories, or try running npm ERR! the command again as root/Administrator.
Oct 25 19:27:52 GET /skServer/debugKeys 200 1.127 ms - -
Oct 25 19:27:52 ENOENT: no such file or directory, open 'c:\signalk\nodejs\node_modules\signalk-server\samples\aava-n2k.data'

```

**Comment by tkurki** [comment]
So the fix would be to close all streams/filestreams before starting the actual update?

**Comment by cmotelet** [comment]
closed via #1172
